### DISCUSSION TITLE
Why Were Pointer Event Names Changed?

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#7907] [Q&A]
Avalonia has the following primary pointer events:

PointerEnter 
PointerLeave 
PointerMoved
PointerPressed 
PointerReleased

UWP, which was the basis for this implementation, has:

PointerEntered
PointerExited
PointerMoved
PointerPressed
PointerReleased
PointerCanceled
PointerCaptureLost

The first observation is a good simplification. UWP has PointerCanceled, PointerCaptureLost and PointerExited which are all used in slightly different situations. UWP apps must actually handle all of these events as PointerExited may never fire after PointerEntered. My assumption is Avalonia cleaned all of this up and fires PointerEnter and PointerLeave consistently doing away with the other two.

However, then we get into naming:

UWP PointerEntered vs Avalonia PointerEnter. Almost all events are past-tense and almost ALWAYS end in 'ed'. This was changed in Avalonia so it no longer matches the other event names or standard convention. In my opinion PointerEnter is misnamed and should be PointerEntered like in UWP.

Then we have PointerLeave. This was probably renamed because it no longer works exactly like UWP -- as mentioned it handles all cases where pointer exits the control for whatever reason. however, again it doesn't end in 'ed' and isn't well paired with the term 'enter'. In English the terms are matched "Enter"/"Exit" (most common) and "Arrive"/"Leave" (slightly less common) as antonyms. So in my optionin this never should have been changed and should have retained the UWP naming PointerExited.

Bottom line, why was this naming changed? And should it be changed back?

@kekekeks

**Comment by kekekeks** [comment]
No idea, naming was here before I've joined. It dates back to pre-0.1.0-alpha release

https://github.com/AvaloniaUI/Avalonia/blob/5e1a282886f0be4a1a36b31788543483debac5db/Perspex/Controls/Control.cs#L83-L87

@grokys

**Reply by robloo** [reply]
Ah, ok, my mistake -- I thought you implemented this with the multi-touch support.

**Comment by robloo** [comment]
@grokys Would there be any objection to a PR that changed events as follows:

 1. PointerEnter -> PointerEntered
 1. PointerLeave -> PointerExited

I am willing to submit the PR to standardize this API. It would be good to do for 11.0.

**Reply by grokys** [reply]
Yeah I think I would support this.
